### PR TITLE
Add option to pass an ELF file to the plugin instead of an array of BPF byte code

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add option to pass an ELF file to the plugin instead of an array of BPF byte code.

This is a prerequisite for fixing #59 as we need to pass maps and relocation info to the plugin.

### CODE REVIEW

File: include/bpf_conformance.h [hunk 1]
```diff
@@ -55,4 +83,13 @@ bpf_conformance(
     bpf_conformance_test_CPU_version_t CPU_version = bpf_conformance_test_CPU_version_t::v3,
     bpf_conformance_list_instructions_t list_instructions_option =
         bpf_conformance_list_instructions_t::LIST_INSTRUCTIONS_NONE,
-    bool debug = false);
+    bool debug = false)
+{
+    bpf_conformance_options_t options;
+    options.include_test_regex = include_test_regex;
+    options.exclude_test_regex = exclude_test_regex;
+    options.CPU_version = CPU_version;
```
Comment (dthaler) [inline include/bpf_conformance.h:91]: The casing of CPU seems anachronistic.  Shouldn't it be "cpu_version"?

File: src/bpf_writer.cc [hunk 2]
```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include <elfio/elfio.hpp>
+
+#include "bpf_writer.h"
+
+const std::string symbol_section_name = ".symtab";
+const std::string maps_section_name = "maps";
```
Comment (dthaler) [inline src/bpf_writer.cc:9]: Be aware that the "maps" section is considered obsolete on Linux, whereas the BTF ".maps" section is current.
https://github.com/dthaler/ebpf-docs-1/blob/elf/rst/elf.rst#data-sections talks about this, but is still missing the details on the ".maps" section.

File: src/ebpf.h [hunk 3]
```diff
@@ -217,4 +217,56 @@ struct ebpf_inst
...
+    BPF_MAP_TYPE_PERCPU_HASH = 4,      ///< Per-CPU hash table.
+    BPF_MAP_TYPE_PERCPU_ARRAY = 5,     ///< Per-CPU array.
+    BPF_MAP_TYPE_HASH_OF_MAPS = 6,     ///< Hash table, where the map value is another map.
+    BPF_MAP_TYPE_ARRAY_OF_MAPS = 7,    ///< Array, where the map value is another map.
+    BPF_MAP_TYPE_LRU_HASH = 8,         ///< Least-recently-used hash table.
+    BPF_MAP_TYPE_LPM_TRIE = 9,         ///< Longest prefix match trie.
+    BPF_MAP_TYPE_QUEUE = 10,           ///< Queue.
+    BPF_MAP_TYPE_LRU_PERCPU_HASH = 11, ///< Per-CPU least-recently-used hash table.
+    BPF_MAP_TYPE_STACK = 12,           ///< Stack.
+    BPF_MAP_TYPE_RINGBUF = 13          ///< Ring buffer.
```
Comment (dthaler) [inline src/ebpf.h:236]: These numbers are Windows numbers and are different on Linux.  Is this what you intend?

File: src/bpf_writer.cc (hunk 2 above)
Comment (Alan-Jowett) [reply src/bpf_writer.cc:11]: Added note and issue to track this.

**Pull request merged**